### PR TITLE
Move all "clades to model" logic to the hub

### PR DESCRIPTION
**Pull request by bsweger**
### Background

A prior version of [get_clades_to_model.py ](https://github.com/reichlab/variant-nowcast-hub/blob/main/src/get_clades_to_model.py) was invoking a [get_clade_list.py script in the cladetime package](https://github.com/reichlab/cladetime/blob/main/src/cladetime/get_clade_list.py).

Splitting the logic between two repos was never a great idea (because getting the clade list is a hub concern, not a cladetime concern). However, some of the metadata maneuvering was so tightly coupled to cladetime that this pattern was expedient at the time.

Now that the CladeTime interface makes it easier to work with sequence metadata, we can completely de-couple this hub-specific logic from cladetime and keep all of the "get clade list" code within the hub repo, where it logically belongs.

#### Logs from `uv run get_clades_to_model.py` on the main branch:

```
Reading inline script metadata from: get_clades_to_model.py
 Updated https://github.com/reichlab/cladetime (5847385)
   Built cladetime @ git+https://github.com/reichlab/cladetime@584738567164d525456cc57903495a74dd6c71b2
Installed 35 packages in 435ms
2024-10-23 18:54:06 [info     ] starting genome metadata download destination=/Users/rsweger/covid_variant/2024-10-23-metadata.tsv.zst source=https://nextstrain-data.s3.amazonaws.com/files/ncov/open/metadata.tsv.zst?versionId=u6mIGkSoh4DO.yk4la1j8lVzDukzqe6K
2024-10-23 18:55:25 [info     ] 'download_covid_genome_metadata' complete elapsed_seconds=79.36
2024-10-23 18:55:48 [info     ] 'get_clades' complete          elapsed_seconds=23.02
10/23/2024 02:55:48 PM -  INFO - Clade list: ['24A', '24B', '24C', '24D', '24E', '24F', '24G', 'recombinant', 'other']
10/23/2024 02:55:49 PM -  INFO - Ncov metadata: {'schema_version': 'v1', 'nextclade_version': 'nextclade 3.9.0', 'nextclade_dataset_name': 'SARS-CoV-2', 'nextclade_dataset_version': '2024-10-17--16-48-48Z', 'nextclade_tsv_sha256sum': 'b3f6e47bb2edd67e296e9e148ee5e8002a13979714fd3a8ed1b68924fda2c75a', 'metadata_tsv_sha256sum': '1b718d192d0e2f3ce31fc85f647e4b610d53a8bba4033e4b21cde37cfd7ca9ec', 'metadata_version_url': 'https://nextstrain-data.s3.amazonaws.com/files/ncov/open/metadata_version.json?versionId=5Qg3oqpJO4HdhgpuRL0Rj0.tdxZNpJZj'}
10/23/2024 02:55:49 PM -  INFO - Clade list saved: /Users/rsweger/code/variant-nowcast-hub/auxiliary-data/modeled-clades/2024-10-30.json
```

#### Logs from `uv run get_clades_to_model.py` on this feature branch:

```
Reading inline script metadata from: get_clades_to_model.py
 Updated https://github.com/reichlab/cladetime (5847385)
10/23/2024 03:15:49 PM -  INFO - Getting clade list
10/23/2024 03:16:11 PM -  INFO - Clade list: ['24A', '24B', '24C', '24D', '24E', '24F', '24G', 'recombinant', 'other']
10/23/2024 03:16:11 PM -  INFO - Ncov metadata: {'schema_version': 'v1', 'nextclade_version': 'nextclade 3.9.0', 'nextclade_dataset_name': 'SARS-CoV-2', 'nextclade_dataset_version': '2024-10-17--16-48-48Z', 'nextclade_tsv_sha256sum': 'b3f6e47bb2edd67e296e9e148ee5e8002a13979714fd3a8ed1b68924fda2c75a', 'metadata_tsv_sha256sum': '1b718d192d0e2f3ce31fc85f647e4b610d53a8bba4033e4b21cde37cfd7ca9ec', 'metadata_version_url': 'https://nextstrain-data.s3.amazonaws.com/files/ncov/open/metadata_version.json?versionId=5Qg3oqpJO4HdhgpuRL0Rj0.tdxZNpJZj'}
10/23/2024 03:16:11 PM -  INFO - Clade list saved: /Users/rsweger/code/variant-nowcast-hub/auxiliary-data/modeled-clades/2024-10-30.json
```

#### Output file created by `uv run get_clades_to_model.py` on this feature branch:
[2024-10-30.json](https://github.com/user-attachments/files/17497323/2024-10-30.json)

### CODE REVIEW

File: src/get_clades_to_model.py [hunk 1]
```diff
@@ -57,21 +59,84 @@ def get_next_wednesday(starting_date: datetime) -> str:
     return next_wednesday.strftime("%Y-%m-%d")
 
 
-def main(round_id: str, clade_output_path: Path):
+def get_clades(
```
Comment (bsweger) [inline src/get_clades_to_model.py:62]: The `get_clades` is [copied from cladetime](https://github.com/reichlab/cladetime/blob/main/src/cladetime/get_clade_list.py#L24). Once this PR is merged, we'll delete the cladetime version.

File: src/get_clades_to_model.py [hunk 2]
```diff
@@ -57,21 +59,84 @@ def get_next_wednesday(starting_date: datetime) -> str:
...
+
+    variants = high_prev_variants.get_column("clade").to_list()[:max_clades]
+
+    return variants
+
+
+def main(
+    round_id: str,
+    clade_output_path: Path,
+    threshold: float = 0.01,
```
Comment (bsweger) [inline src/get_clades_to_model.py:116]: Threshold defaults [copied from cladetime](https://github.com/reichlab/cladetime/blob/main/src/cladetime/get_clade_list.py#L71)

**Review by rogersbw — APPROVED** [review]
Seems reasonable and properly working.

**Comment by bsweger** [comment]
Also ran the "open round" workflow for good measure: https://github.com/reichlab/variant-nowcast-hub/pull/124

**Pull request merged**